src/lxc/lsm/apparmor.c

```diff
@@ -159,61 +487,578 @@ static bool aa_needs_transition(char *curlabel)
 	return true;
 }
 
+static inline void uint64hex(char *buf, uint64_t num)
+{
+	size_t i;
+
+	buf[16] = 0;
+	for (i = 16; i--;) {
+		char c = (char)(num & 0xf);
+		buf[i] = c + (c < 0xa ? '0' : 'a' - 0xa);
+		num >>= 4;
+	}
+}
+
+static inline char *shorten_apparmor_name(char *name)
+{
+	size_t len = strlen(name);
+	if (len + 7 > 253) {
+		uint64_t hash;
+		hash = fnv_64a_buf(name, len, FNV1A_64_INIT);
+		name = must_realloc(name, 16+1);
+		uint64hex(name, hash);
+	}
+
+	return name;
+}
+
+/* Replace slashes with hyphens */
+static inline void sanitize_path(char *path)
+{
+	size_t i;
+
+	for (i = 0; path[i]; ++i)
```

Comment:
Nit: We usually use `i++` across the codebase unless it's an important side-effect that we need.